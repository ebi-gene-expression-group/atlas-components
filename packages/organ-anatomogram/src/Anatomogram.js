import React from 'react'
import PropTypes from 'prop-types'

import Switcher from './Switcher'
import AnatomogramSvg from './AnatomogramSvg'
import {getDefaultView, getParentView, supportedSpecies} from './Assets'

class Anatomogram extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      selectedView: getDefaultView(props.species),
      parentView: getParentView(props.species, getDefaultView(props.species))
    }
    this._switchAnatomogramView = this._switchAnatomogramView.bind(this)
  }

  componentDidUpdate(prevProps) {
    if (this.props.species !== prevProps.species) {
      this.setState({
        selectedView: getDefaultView(this.props.species),
        parentView: getParentView(this.props.species, getDefaultView(this.props.species))
      })
    }
  }

  _switchAnatomogramView(species, anatomogramView) {
    this.props.onClick([])
    this.setState({
      selectedView: anatomogramView,
      parentView: getParentView(species, anatomogramView),
      showIds: [],
      selectIds: [],
      highlightIds:[]
    }, this.props.afterSwitchView(species, anatomogramView))
  }

  static getDerivedStateFromProps(props, state) {
    if (props.species !== state.species) {
      return {
        species: props.species,
        selectedView: getDefaultView(props.species)
      }
    }
    return null
  }

  render() {
    const { parentView, selectedView } = this.state
    const { species, showLinkBoxIds } = this.props
    return (
      supportedSpecies.includes(species) &&
        <div>
          <Switcher
            species={species}
            parentView={parentView}
            selectedView={selectedView}
            onChangeView={this._switchAnatomogramView} />

          <AnatomogramSvg
            {...this.props}
            showLinkBoxIds={showLinkBoxIds}
            onChangeView={this._switchAnatomogramView}
            selectedView={selectedView} />
        </div>
    )
  }
}

Anatomogram.propTypes = {
  species: PropTypes.string.isRequired,
  showLinkBoxIds: PropTypes.func.isRequired
}

export default Anatomogram
