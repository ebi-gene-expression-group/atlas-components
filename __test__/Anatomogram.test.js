import React from 'react'
import { shallow } from 'enzyme'

import Anatomogram from '../src/Anatomogram'

describe(`Anatomogram`, () => {

  test(`should render for a supported species`, () => {
    expect(shallow(<Anatomogram
      idsWithMarkup={[]}
      onClick={jest.fn()}
      onMouseOut={jest.fn()}
      onMouseOver={jest.fn()}
      parentView={`parent`}
      species={`mus_musculus`}
      initShowIds={() => {}}
      clearSelectIds={() => {}}
    />)).toMatchSnapshot()
  })

  test(`should not render for an unsupported species`, () => {
    expect(shallow(<Anatomogram species={`ovis_aries`}
      initShowIds={() => {}} clearSelectIds={() => {}}/>).children()).toHaveLength(0)
  })

})
