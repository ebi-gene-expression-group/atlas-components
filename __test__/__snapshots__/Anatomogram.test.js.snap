// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anatomogram should render for a supported species 1`] = `
<div>
  <Switcher
    onChangeView={[Function]}
    selectedView="male"
    species="mus_musculus"
  />
  <AnatomogramSvg
    clearSelectIds={[Function]}
    idsWithMarkup={Array []}
    initShowIds={[Function]}
    onChangeView={[Function]}
    onClick={[MockFunction]}
    onInjected={[Function]}
    onMouseOut={[MockFunction]}
    onMouseOver={[MockFunction]}
    parentView="parent"
    selectedView="male"
    species="mus_musculus"
  />
</div>
`;
